libs/dal/src/repositories/topic/topic.repository.ts

```diff
@@ -30,14 +31,41 @@ export class TopicRepository extends BaseRepository<EnforceEnvironmentQuery, Top
   }
 
   async findTopic(entity: Omit<TopicEntity, 'key' | 'name'>): Promise<TopicEntity> {
+    const TOPIC_SUBSCRIBERS_COLLECTION = 'topicsubscribers';
+
     const { _environmentId, _id, _organizationId, _userId } = entity;
 
-    return await this.findOne({
-      _environmentId,
-      _id,
-      _organizationId,
-      _userId,
-    });
+    const [result] = await this.aggregate([
+      {
+        $match: { _organizationId, _environmentId, _id, _userId },
+      },
+      {
+        $lookup: {
+          from: TOPIC_SUBSCRIBERS_COLLECTION,
+          localField: '_id',
+          foreignField: '_topicId',
+          as: 'topicSubscribers',
+        },
+      },
+      {
+        $project: {
+          _environmentId: 1,
+          _organizationId: 1,
+          _userId: 1,
+          key: 1,
+          name: 1,
+          // The lookup returns a matrix so we return the first array element in the projection
+          subscribers: { $arrayElemAt: ['$topicSubscribers.subscribers', 0] },
```

Comment:
what about the `customData` field?

Reply:
Is something out of scope for now as we won't implement it in this cycle so I'd rather not return it.

Reply:
should we then keep it on the entity?

Reply:
I can't elaborate a strong argument in favour of keeping it in the entity, that's true. Up to what the team decides. To leave it as an indicator of next steps (with links to the documentation) or remove it.